x-pack/plugin/analytics/src/main/java/org/elasticsearch/xpack/analytics/mapper/HistogramFieldMapper.java

```diff
@@ -447,4 +451,74 @@ public int count() {
             return count;
         }
     }
+
+    @Override
+    public SourceLoader.SyntheticFieldLoader syntheticFieldLoader() {
+        if (ignoreMalformed.value()) {
+            throw new IllegalArgumentException(
+                "field [" + name() + "] of type [histogram] doesn't support synthetic source because it ignores malformed histograms"
+            );
+        }
+        if (copyTo.copyToFields().isEmpty() != true) {
+            throw new IllegalArgumentException(
+                "field [" + name() + "] of type [histogram] doesn't support synthetic source because it declares copy_to"
+            );
+        }
+        return new SourceLoader.SyntheticFieldLoader() {
+            private final InternalHistogramValue value = new InternalHistogramValue();
+            private BytesRef binaryValue;
+
+            @Override
+            public Stream<Map.Entry<String, StoredFieldLoader>> storedFieldLoaders() {
+                return Stream.of();
+            }
+
+            @Override
+            public DocValuesLoader docValuesLoader(LeafReader leafReader, int[] docIdsInLeaf) throws IOException {
+                BinaryDocValues docValues = DocValues.getBinary(leafReader, fieldType().name());
+                if (docValues == null) {
```

Comment:
`DocValues.getBinary()` will return an empty impl rather than null.  Call `leafReader.getBinary()` instead?

Reply:
Which makes me think, do we have a base test that checks that an index with synthetic source enabled will return `null` here when no documents contain a given field? I think that would be helpful in detecting these sort of issues and might catch performance regressions too.

Reply:
We make sure that we spit out nothing but we don't detect the null. I'll have a look.